into/backends/aws.py

```diff
@@ -0,0 +1,248 @@
+from __future__ import print_function, division, absolute_import
+
+import os
+import uuid
+
+from contextlib import contextmanager
+from collections import Iterator
+
+try:
+    from urlparse import urlparse
+except ImportError:
+    from urllib.parse import urlparse
+
+import pandas as pd
+from toolz import memoize
+
+import boto
+
+from into import discover, CSV, resource, append, convert, drop, Temp, JSON
+from into import JSONLines, SSH, into, chunks, HDFS
+
+from .text import TextFile
+from ..utils import tmpfile, ext, sample
+
+
+@memoize
+def get_s3_connection(aws_access_key_id=None,
+                      aws_secret_access_key=None,
+                      anon=False):
+    cfg = boto.Config()
+
+    if aws_access_key_id is None:
+        aws_access_key_id = cfg.get('Credentials', 'aws_access_key_id')
+
+    if aws_access_key_id is None:
+        aws_access_key_id = os.environ.get('AWS_ACCESS_KEY_ID')
+
+    if aws_secret_access_key is None:
+        aws_secret_access_key = cfg.get('Credentials', 'aws_secret_access_key')
+
+    if aws_secret_access_key is None:
+        aws_secret_access_key = os.environ.get('AWS_SECRET_ACCESS_KEY')
+
+    # anon is False but we didn't provide any credentials so try anonymously
+    anon = (not anon and
+            aws_access_key_id is None and
+            aws_secret_access_key is None)
+    return boto.connect_s3(aws_access_key_id, aws_secret_access_key,
+                           anon=anon)
+
+
+class _S3(object):
+    def __init__(self, uri, s3=None, aws_access_key_id=None,
+                 aws_secret_access_key=None, *args, **kwargs):
+        result = urlparse(uri)
+        self.bucket = result.netloc
+        self.key = result.path.lstrip('/')
+
+        if s3 is not None:
+            self.s3 = s3
+        else:
+            self.s3 = get_s3_connection(aws_access_key_id=aws_access_key_id,
+                                        aws_secret_access_key=aws_secret_access_key)
+        try:
+            bucket = self.s3.get_bucket(self.bucket)
+        except boto.exception.S3ResponseError:
+            bucket = self.s3.create_bucket(self.bucket)
+
+        self.object = bucket.get_key(self.key)
+        if self.object is None:
+            self.object = bucket.new_key(self.key)
+
+        self.subtype.__init__(self, uri, *args, **kwargs)
+
+
+@memoize
+def S3(cls):
+    """Parametrized S3 bucket Class
+
+    Examples
+    --------
+    >>> import sqlalchemy
+    >>> S3(CSV)
+    S3(CSV)
+    >>> S3(sqlalchemy.Table)
```

Comment:
Can one put a `sqlalchemy.Table` on S3?  Perhaps also show how to create an `S3(CSV)` object explicitly

    >>> S3(CSV)(what-uri-is-appropriate?,  what kwargs should I know about?)